### ISSUE TITLE
When mix of fixable and in-fixable error,  `stdout` will throw error.

### ISSUE THREAD

**Issue by RayGuo-ergou** [mantoni/eslint_d.js#309]
Hi I have upgrade to v14.0.2 and found that if I have a in-fixable rule e.g. `no-console` , a error will throw.

For example for this code

```ts
/* eslint no-console: "error" */
  console.log(2)
```

I would except to have to fix as
```ts
console.log(2)
```

<hr />

Here's the log/error I got
> 22:25:40[ERROR] Formatter 'eslint_d' error: console.log(2)

**Comment by RayGuo-ergou** [comment]
```
          22:34:55[DEBUG] Running formatters on /home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts: { "eslint_d" }
          22:34:55[INFO] Run eslint_d on /home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts
          22:34:55[DEBUG] Run command: { "eslint_d", "--fix-to-stdout", "--stdin", "--stdin-filename", "/home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts" }
          22:34:55[DEBUG] Run CWD: /home/raydev/git/userscripts
          22:34:55[INFO] eslint_d exited with code 1
          22:34:55[DEBUG] eslint_d stdout: { "console.log(2)", "" }
          22:34:55[DEBUG] eslint_d stderr: { "" }
          22:34:55[ERROR] Formatter 'eslint_d' error: console.log(2)

```

**Comment by RayGuo-ergou** [comment]
It seems intentional 🤔?

https://github.com/mantoni/eslint_d.js/blob/master/test/test.integration.js#L261

**Comment by RayGuo-ergou** [comment]
I found where's been set, do you agree to change to something like

```diff
      // The remaining 'content' must be the termination code:
      const match = content.match(EXIT_TOKEN_REGEXP);
      if (match) {
+        if (!stderr){
+          process.exitCode = 0;
+          return;
+        }
        process.exitCode = Number(match[1]);
        return;
      }
```

**Comment by RayGuo-ergou** [comment]
I found where it's been set,

Do you agree to change to something like

```diff
-  const { stdout } = supportsColor;
+  const { stdout, stderr } = supportsColor;

      // The remaining 'content' must be the termination code:
      const match = content.match(EXIT_TOKEN_REGEXP);
      if (match) {
+        if (!stderr){
+          process.exitCode = 0;
+          return;
+        }
        process.exitCode = Number(match[1]);
        return;
      }
```

**Comment by RayGuo-ergou** [comment]
Do you agree change to something like this?

```diff
      // The remaining 'content' must be the termination code:
      const match = content.match(EXIT_TOKEN_REGEXP);
      if (match) {
+        if (!process.stderr.read()){
+          process.exitCode = 0;
+          return;
+        }
        process.exitCode = Number(match[1]);
        return;
      }
```

**Comment by DamienCassou** [comment]
> It seems intentional 🤔?

yes. eslint's interface is to return 0 when there is no linting issue and 1 if there is, even when using `--fix-dry-run`. If we see eslint_d as a thin wrapper around eslint to make it faster, then I think eslint_d should follow what eslint is doing.

But `--fix-to-stdout` is not part of eslint's interface so there is some freedom here. If we see `--fix-to-stdout` as a command to only fix things, returning 0 in the presence of unfixable errors is the way to go.

> Looking at the referenced test case, I don't see a difference to the test before this one. I guess there's a subtle difference I'm missing?

I don't understand the question but I hope the above will have answered it anyway :-).

Here is a possible patch if we want to go this way:

```patch
From 3e2851a5eb38dd8831f6416ec5717e2e74a833e2 Mon Sep 17 00:00:00 2001
From: Damien Cassou <damien@cassou.me>
Date: Fri, 2 Aug 2024 20:46:16 +0200
Subject: [PATCH] Make --fix-to-stdout return 0 even in the presence of
 unfixable errors

---
 lib/forwarder.js         | 20 ++++++++++----------
 lib/forwarder.test.js    |  2 +-
 test/test.integration.js |  6 ++----
 3 files changed, 13 insertions(+), 15 deletions(-)

diff --git a/lib/forwarder.js b/lib/forwarder.js
index a60edb3..626afbd 100644
--- a/lib/forwarder.js
+++ b/lib/forwarder.js
@@ -69,24 +69,24 @@ export async function forwardToDaemon(resolver, config) {
         try {
           const { output } = JSON.parse(flushMessage())[0];
           process.stdout.write(output || text);
+          process.exitCode = 0;
         } catch (err) {
           process.stdout.write(text);
           console.error(`eslint_d: ${err}`);
           process.exitCode = 1;
+        }
+      } else {
+        // The remaining 'content' must be the termination code:
+        const match = content.match(EXIT_TOKEN_REGEXP);
+        if (match) {
+          process.exitCode = Number(match[1]);
           return;
         }
-      }
 
-      // The remaining 'content' must be the termination code:
-      const match = content.match(EXIT_TOKEN_REGEXP);
-      if (match) {
-        process.exitCode = Number(match[1]);
-        return;
+        process.stdout.write(content);
+        console.error('eslint_d: unexpected response');
+        process.exitCode = 1;
       }
-
-      process.stdout.write(content);
-      console.error('eslint_d: unexpected response');
-      process.exitCode = 1;
     })
     .on('error', async (err) => {
       if (err['code'] === 'ECONNREFUSED') {
diff --git a/lib/forwarder.test.js b/lib/forwarder.test.js
index 0829cb4..5f67ccd 100644
--- a/lib/forwarder.test.js
+++ b/lib/forwarder.test.js
@@ -277,7 +277,7 @@         forwardToDaemon(resolver, config);
         socket.on.secondCall.callback(); // end
 
         assert.calledWith(process.stdout.write, 'response from daemon');
-        assert.equals(process.exitCode, 1);
+        assert.equals(process.exitCode, 0);
         refute.called(console.error);
       });
 
diff --git a/test/test.integration.js b/test/test.integration.js
index 34bb33d..6f645ed 100644
--- a/test/test.integration.js
+++ b/test/test.integration.js
@@ -238,8 +238,7 @@           it('prints input if no change is needed', async () => {
               `/* eslint radix: "error" */
             console.log('Hello' + parseInt('087'))`
             );
-            refute.isNull(error);
-            assert.equals(error?.['code'], 1);
+            assert.isNull(error);
           });
 
           it('prints fixed output if change is needed', async () => {
@@ -257,8 +256,7 @@           it('prints fixed output if change is needed', async () => {
               `/* eslint radix: "error" */
             console.log('Hello' + parseInt('087'))`
             );
-            refute.isNull(error);
-            assert.equals(error?.['code'], 1);
+            assert.isNull(error);
           });
         });
       });
-- 
2.45.2

```

**Comment by mantoni** [comment]
📦 Fix released in `v14.0.3`. Thank you all ✨

**Comment by RayGuo-ergou** [comment]
Thank you!
